src/components/ProductAnalytics/useAnalytics.ts

```diff
@@ -13,23 +14,21 @@ interface Analytics {
 
 export function useAnalytics(): Analytics {
   const posthog = usePostHog();
-  const [hasBeenUserSet, setHasBeenUserSet] = useLocalStorage<boolean>(
-    "analyticsHasBeenUserSet",
-    false,
-  );
+
+  const { register } = useRouteChange(location => {
+    trackEvent("$pageview", {
+      normalized_path: location.pathname,
+    });
+  });
 
   function initialize(userProperties: UserProperties) {
     if (!posthog) return;
 
-    // initialize function is called on each reload that cause generates new used id by identify function
-    // to avoid this we need to check if user has been set
-    if (hasBeenUserSet) return;
+    register();
 
     const id = posthog.get_distinct_id();
 
     posthog.identify(id, userProperties);
```

Comment:
suggestion: now identify and underhood Set will be fired each time the user reloads the browser

Reply:
yes